samples/template-operator/operator/controllers/sample_controller.go

```diff
@@ -85,20 +109,118 @@ func (r *SampleReconciler) HandleInitialState(ctx context.Context, sampleResourc
 	return r.Client.Status().Update(ctx, sampleResource)
 }
 
-func (r *SampleReconciler) HandleProcessingState(ctx context.Context, sampleResource *v1alpha1.Sample) error {
+func (r *SampleReconciler) HandleProcessingState(ctx context.Context, sampleResource *v1alpha1.Sample, logger *logr.Logger) error {
 	// TODO: processing logic here
 
-	// Example: If Processing succeeds, set to Ready state
-	sampleResource.Status.State = v1alpha1.SampleStateReady
-	return r.Client.Status().Update(ctx, sampleResource)
+	sampleObjUnstructured, err := runtime.DefaultUnstructuredConverter.ToUnstructured(sampleResource)
+	if err != nil {
+		return err
+	}
+
+	manifestClient, err := r.getManifestClient(logger, sampleResource.Spec.ChartFlags)
+	if err != nil {
+		sampleResource.Status.State = v1alpha1.SampleStateError
+		return r.Client.Status().Update(ctx, sampleResource)
+	}
+
+	// Use manifest library client to install a sample chart
+	ready, err := manifestClient.Install(manifestLib.InstallInfo{
+		Ctx: ctx,
+		ChartInfo: &manifestLib.ChartInfo{
+			ChartPath:   sampleResource.Spec.ChartPath,
+			ReleaseName: sampleResource.Spec.ReleaseName,
+		},
+		RemoteInfo: custom.RemoteInfo{
+			// destination cluster rest config
+			RemoteConfig: r.Config,
+			// destination cluster rest client
+			RemoteClient: &r.Client,
+		},
+		ResourceInfo: manifestLib.ResourceInfo{
+			// base operator resource to be passed for custom checks
+			BaseResource: &unstructured.Unstructured{Object: sampleObjUnstructured},
+		},
+		CheckFn: func(context.Context, *unstructured.Unstructured, *logr.Logger, custom.RemoteInfo) (bool, error) {
+			// your custom logic here to set ready state
+			return true, nil
+		},
+		CheckReadyStates: false,
```

Comment:
I think we should default this to true since most people want to have a ready check, probably we can use a similar default readyness as the declarative pattern with services/deployments etc

Reply:
this verifies ready state for all 'native' installed resources, supported by helm. I will enable it.